server/tests/riders.test.ts

```diff
@@ -0,0 +1,261 @@
+  describe('GET /api/riders/abc-10/favorites', () => {
+    it('should return 200 OK', async () => {
+      const res = await request(app)
+        .get('/api/riders/abc-10/favorites')
+        .set('Authorization', `Bearer ${adminToken}`)
+        .expect(200)
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      expect(res.status).to.be.equal(200);
+      expect(res.body).to.have.property('data');
+      expect(res.body.data).to.deep.equal(riders[0].favoriteLocations);
+    });
+  });
+
+  const sendRidersData = {
+    email: 'test-email2@test.com',
+    phoneNumber: '1234567892',
+    firstName: 'Test',
+    lastName: 'Testing2',
+    pronouns: 'he/him/his',
+    accessibility: 'Crutches',
+    description: '',
+    joinDate: '2023-03-09',
+    endDate: '2024-03-09',
+    address: '3 Colonial Ln, Ithaca, NY 14850',
+    favoriteLocations: ['west'],
+    organization: 'CULift',
+    photoLink: '',
+    active: true,
+  };
+
+  // testing the addition of a new rider
+  describe('POST /api/riders', () => {
+    it('should return 200 OK', async () => {
+      const res = await request(app)
+        .post('/api/riders/')
+        .send(sendRidersData)
+        .set('Authorization', `Bearer ${adminToken}`)
+        .expect(200)
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      const sentData = { ...res.body.data };
+      // this is randomly generated and cannot be tested for,
+      // but the accuracy of the rest of the data can be
+      delete sentData.id;
+      expect(sentData).to.deep.equal(sendRidersData);
+    });
+  });
+
+  // testing the addition of a favorite location for a rider
+  // ***************************************************
+  // getting a 400 bad request for some reason
+  // ***************************************************
+  describe('POST /api/riders/abc-10/favorites', () => {
+    it('should return 200 OK', async () => {
+      const res = await request(app)
+        .post('/api/riders/abc-10/favorites')
+        .send({ id: 'central' })
+        .set('Authorization', `Bearer ${adminToken}`)
+        .expect(200)
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      expect(res.body.id).to.deep.equal(
+        riders[0].favoriteLocations[riders[0].favoriteLocations.length - 1]
+      );
+    });
+  });
+
+  // testing the updating of info of a rider
+  describe('PUT /api/riders/abc-10', () => {
+    it('should return 200 OK', async () => {
+      const res = await request(app)
+        .put('/api/riders/abc-10')
+        .send({ firstName: 'NewName' })
+        .set('Authorization', `Bearer ${adminToken}`)
+        .expect(200)
+        .expect('Content-Type', 'application/json; charset=utf-8');
+      expect(res.body.data.firstName).to.be.equal('NewName');
+    });
+  });
+
+  // testing the deletion of a rider
+  describe('DELETE /api/riders/abc-10', () => {
+    it('should return 200 OK', async () => {
```

Comment:
check that the object is actually deleted by trying to get the object by id from the model. it should give an error